lib/web/apiserver.go

```diff
@@ -3162,6 +3162,11 @@ func (h *Handler) mfaLoginFinishSession(w http.ResponseWriter, r *http.Request,
 	case errors.Is(err, types.ErrPassswordlessLoginBySSOUser):
 		return nil, trace.Wrap(err)
 
+	// Return a friendlier error if the user has assigned a role that doesn't exist in the
+	// backend.
+	case errors.Is(err, types.ErrNonExistingRoleAssigned):
+		return nil, trace.Wrap(err)
```

Comment:
How is this "returning a friendlier" error? Looks like we're just wrapping the existing error, not doing anything to make it friendlier.

Reply:
If you look at the `case err != nil:` below it returns `trace.AccessDenied("invalid credentials")` to not leak any potentially sensitive information. If we know this is `ErrNonExistingRoleAssigned`, we know it's safe to print its content to an unauthorized (but authenticated) user, which is "friendlier" than printing "invalid credentials".